api/src/main/java/org/pmiops/workbench/cloudtasks/TaskQueueService.java

```diff
@@ -0,0 +1,99 @@
+package org.pmiops.workbench.cloudtasks;
+
+import com.google.cloud.tasks.v2.AppEngineHttpRequest;
+import com.google.cloud.tasks.v2.CloudTasksClient;
+import com.google.cloud.tasks.v2.HttpMethod;
+import com.google.cloud.tasks.v2.QueueName;
+import com.google.cloud.tasks.v2.Task;
+import com.google.gson.Gson;
+import com.google.protobuf.ByteString;
+import java.util.List;
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.config.WorkbenchLocationConfigService;
+import org.pmiops.workbench.model.AuditProjectAccessRequest;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TaskQueueService {
+  private static final String BASE_PATH = "/v1/cloudTask";
+  private static final String EXPORT_RESEARCHER_PATH = BASE_PATH + "/exportResearcherData";
+  private static final String EXPORT_WORKSPACE_PATH = BASE_PATH + "/exportWorkspaceData";
+  private static final String AUDIT_PROJECTS_PATH = BASE_PATH + "/auditProjectAccess";
+
+  private static final String AUDIT_PROJECTS_QUEUE_NAME = "auditProjectQueue";
+
+  private WorkbenchLocationConfigService locationConfigService;
+  private Provider<CloudTasksClient> cloudTasksClientProvider;
+  private Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  public TaskQueueService(
+      WorkbenchLocationConfigService locationConfigService,
+      Provider<CloudTasksClient> cloudTasksClientProvider,
+      Provider<WorkbenchConfig> configProvider) {
+    this.locationConfigService = locationConfigService;
+    this.cloudTasksClientProvider = cloudTasksClientProvider;
+    this.workbenchConfigProvider = configProvider;
+  }
+
+  public void groupAndPushRdrWorkspaceTasks(List<Long> workspaceIds) {
+    groupAndPushRdrWorkspaceTasks(workspaceIds, false);
+  }
+
+  public void groupAndPushRdrWorkspaceTasks(List<Long> workspaceIds, boolean backfill) {
```

Comment:
These are really nice. Do you think further parameterization be worth the effort?

Reply:
Possibly - it feels there is still some duplication here, but I want to be careful about overfitting to the current patterns to avoid implicit dependencies. The generic concept is "split this list of IDs into buckets, and dispatch them to a given endpoint". The details that might diverge here are the request payload for a task, the type of object in the request, the endpoint, the queue, and the batch size. I think the previous RdrTaskQueue was guilty of this: it's actually hitting two different task endpoints, but forcing the assumption that they accept identical query parameter payloads.

I also don't like the parameter pattern, and I switched to a different pattern with the audit tasks - i.e. using a specific extensible request payload, rather than a raw array of ids.

Reply:
Reading your comment again - I'm not sure whether I understood your question correctly. Let me know if my answer addressed it.